### PR TITLE
[DataGrid] Allow to override the default overlay height in `autoHeight` mode

### PR DESCRIPTION
**Pull request by cherniavskii**
Solves https://github.com/mui/mui-x/issues/5811#issuecomment-1685762082

Preview: https://deploy-preview-10203--material-ui-x.netlify.app/x/react-data-grid/layout/#overlay-height

TODO:
- [x] Add unit test
- [x] Avoid creating a screenshot for the new demo?

### CODE REVIEW

**Review by romgrk — APPROVED** [review]

File: packages/grid/x-data-grid/src/hooks/features/rows/gridRowsUtils.ts [hunk 1]
```diff
@@ -389,5 +389,5 @@ export function getMinimalContentHeight(
   rowHeight: number,
 ) {
   const densityFactor = gridDensityFactorSelector(apiRef);
-  return 2 * Math.floor(rowHeight * densityFactor);
+  return `var(--DataGrid-noRowsOverlayHeight, ${2 * Math.floor(rowHeight * densityFactor)}px)`;
```
Comment (MBilalShafi) [inline packages/grid/x-data-grid/src/hooks/features/rows/gridRowsUtils.ts:392]: I feel it doesn't strictly relate to the `NoRowsOverlay` rather it applies in general to all the overlays, for instance in this [demo](https://codesandbox.io/s/great-bird-nv953p), I tried to set a height of `500px` and show a loading overlay which also reflects the same height.
```suggestion
  return `var(--DataGrid-overlayHeight, ${2 * Math.floor(rowHeight * densityFactor)}px)`;
```

File: docs/data/data-grid/layout/layout.md [hunk 2]
```diff
@@ -31,6 +31,14 @@ This is not recommended for large datasets as row virtualization will not be abl
 
 {{"demo": "AutoHeightGrid.js", "bg": "inline"}}
 
+### No rows height
+
+By default, when the Data Grid has no rows, the rows container takes the height of two rows.
+
+To adjust the height of the rows container, use the `--DataGrid-noRowsOverlayHeight` CSS variable.
```
Comment (MBilalShafi) [inline docs/data/data-grid/layout/layout.md:38]: In accordance with the other comment.
```suggestion
### Grid overlays' height

By default, the Data Grid overlays like `<NoRowsOverlay />` or `<LoadingOverlay />` take the height of two rows.

To customize the height of the overlay, use the `--DataGrid-overlayHeight` CSS variable.
```

**Review by MBilalShafi — COMMENTED** [review]
CSS variables is a really cool way to allow this customization. Thanks 🎉

File: packages/grid/x-data-grid/src/hooks/features/rows/gridRowsUtils.ts (hunk 1 above)
Comment (cherniavskii) [reply packages/grid/x-data-grid/src/hooks/features/rows/gridRowsUtils.ts:392]: Huh, this was not my intention, but it actually makes sense! 😅
Thanks for spotting it!

**Pull request merged**